gammapy/spectrum/spectrum_dataset.py

```diff
@@ -0,0 +1,205 @@
+
+        self.ONcounts = ONcounts
+        self.OFFcounts = OFFcounts
+        self.livetime = livetime
+        self.mask = mask
+        self.aeff = aeff
+        self.edisp = edisp
+
+        self.model = model
+
+    @property
+    def alpha(self):
+        return self.ONcounts.backscal / self.OFFcounts.backscal
+
+    @property
+    def model(self):
+        return self._model
+
+    @model.setter
+    def model(self, model):
+        self._model = model
+        if model is not None:
+            self._parameters = Parameters(self._model.parameters.parameters)
+            if self.edisp is None:
+                self._predictor = SpectrumEvaluator(model=self.model,
+                                                    livetime=self.livetime,
+                                                    aeff=self.aeff,
+                                                    e_true=self.ONcounts.energy.bins)
+            else:
+                self._predictor = SpectrumEvaluator(model=self.model,
+                                                    aeff=self.aeff,
+                                                    edisp=self.edisp,
+                                                    livetime=self.livetime)
+
+        else:
+            self._parameters = None
+            self._predictor = None
+
+    @property
+    def parameters(self):
+        if self._parameters is None:
+            raise AttributeError("No model set for Dataset")
+        else:
+            return self._parameters
+
+    @property
+    def data_shape(self):
+        """Shape of the counts data"""
+        return self.counts.data.shape
+
+    def npred(self):
+        """Returns npred counts vector """
+        if self._predictor is None:
+            raise AttributeError("No model set for this Dataset")
+        model_npred = self._predictor.compute_npred().data.data
+        return model_npred
+
+    def likelihood_per_bin(self):
+        """Likelihood per bin given the current model parameters"""
+        on_stat_ = wstat(
+            n_on=self.ONcounts.data.data,
+            n_off=self.OFFcounts.data.data,
+            alpha=self.alpha,
+            mu_sig=self.npred(),
+        )
+        return np.nan_to_num(on_stat_)
+
+    def likelihood(self, parameters, mask=None):
+        """Total likelihood given the current model parameters.
+
+        Parameters
+        ----------
+        mask : `~numpy.ndarray`
+            Mask to be combined with the dataset mask.
+        """
+        if self.mask is None and mask is None:
+            stat = self.likelihood_per_bin()
+        elif self.mask is None:
+            stat = self.likelihood_per_bin()[mask]
+        elif mask is None:
+            stat = self.likelihood_per_bin()[self.mask]
+        else:
+            stat = self.likelihood_per_bin()[mask & self.mask]
+
+        return np.sum(stat, dtype=np.float64)
+
+    @classmethod
+    def read_from_ogip(cls, filename):
```

Comment:
Just call this method `.read()`? If we add other I/O formats in future, they'll probably be handled internally and not by multiple `read_...` methods.

Reply:
OK.  I implemented a specific method for OGIP files because the initial plan for `Dataset` I/O was using yaml files. 
I guess I can make `.read_from_ogip()` private and call it from `.read()` so that we can later fill in other I/O methods. Would that be OK? Or would you prefer make it the `read` method and later modify its behavior?

Reply:
Both is fine by me. You could either introduce a private `_read_folder()` and call it from `.read()` or just keep the implementation in `.read()` and we'll modify it later...